Umee-Bots/Large-Borrow/src/agent.ts

```diff
@@ -0,0 +1,48 @@
+import { ethers, Finding, getEthersProvider, HandleTransaction, TransactionEvent } from "forta-agent";
+import BigNumber from "bignumber.js";
+import { BALANCE_OF_ABI, BORROW_ABI, GET_RESERVE_DATA_ABI } from "./constants";
+import { AgentConfig, createFinding, SmartCaller } from "./utils";
+import CONFIG from "./agent.config";
+
+BigNumber.set({ DECIMAL_PLACES: 18 });
+
+export const provideHandleTransaction = (
+  provider: ethers.providers.Provider,
+  config: AgentConfig
+): HandleTransaction => {
+  const LendingPool = SmartCaller.from(
+    new ethers.Contract(config.lendingPoolAddress, [GET_RESERVE_DATA_ABI], provider),
+    {
+      cacheByBlockTag: false,
+    }
+  );
+  const threshold = new BigNumber(config.tvlPercentageThreshold);
+
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const logs = txEvent.filterLog(BORROW_ABI, config.lendingPoolAddress);
+
+    await Promise.all(
+      logs.map(async (log) => {
+        const { reserve, amount, user, onBehalfOf } = log.args;
+
+        const { uTokenAddress } = await LendingPool.getReserveData(reserve, { blockTag: "latest" });
+        const erc20Asset = SmartCaller.from(new ethers.Contract(reserve, [BALANCE_OF_ABI], provider));
+        const tvl = await erc20Asset.balanceOf(uTokenAddress, { blockTag: txEvent.blockNumber - 1 });
+
+        const percentage = new BigNumber(amount.toString()).div(new BigNumber(tvl.toString())).shiftedBy(2);
```

Comment:
`amount` and `tvl` are already `BigNumber` why not using `.div` and `.shiftedBy(2)` directly? 
```
amount.div(tvl).shl(2)
```

Reply:
No, because `ethers.BigNumber` doesn't support decimals directly. So, to do that with `ethers.BigNumber`, we'd need to choose a precision and adapt the division to use it so the final result would be, e.g. `99` instead of `0.99` which would be truncated to `0` (the `shl` would also need to be before the division).

`BigNumber` makes this easier and more straightforward because it supports decimals directly, so just a simple division works and it will automatically show the decimal point when `bn.toString()` is called.